tools/functional-tester/etcd-runner/main.go

```diff
@@ -310,10 +320,24 @@ func runWatcher(eps []string) {
 						return
 					}
 				}
-			}(prefixes[i])
+			}(roundPrefix + "-" + prefixes[i])
 		}
 	}
 	wg.Wait()
+
+	// cancel all watch channels
+	cancelc()
+
```

Comment:
@sinsharat I don't think this is a bug:
`01:51:48.535179 expected wc to be closed, but received {{0 0 0 0} [] 0 false false 0xc048186680}`
That `0xc...` pointer points to an error. The response `Err()` is probably something about the connection closing because there's a `defer rc.Close()` before `cancelc()` gets called. The `cancelc()` call should have no effect because closing the client cancels all associated watchers and posts an error on all outstanding watch channels. This is intentional behavior.